source/adapters/level_zero/queue.cpp

```diff
@@ -1312,13 +1301,9 @@ ur_result_t urQueueReleaseInternal(ur_queue_handle_t Queue) {
   if (!UrQueue->RefCount.decrementAndTest())
     return UR_RESULT_SUCCESS;
 
-  for (auto Cache : UrQueue->EventCaches) {
-    for (auto Event : *Cache) {
+  for (auto &Cache : UrQueue->EventCaches)
+    for (auto &Event : Cache)
       UR_CALL(urEventReleaseInternal(Event));
-    }
-    Cache->clear();
```

Comment:
We should be still clearing the cache at release.

Reply:
Done.